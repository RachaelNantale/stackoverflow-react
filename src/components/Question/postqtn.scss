.post-container {
  padding: 10px;
  margin:20px;
 align-content: center;
 margin: 0 auto;
}
