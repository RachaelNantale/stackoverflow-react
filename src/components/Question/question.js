import React from 'react';
import PropTypes from 'prop-types';
import './postqtn.scss';

const Question = ({ onSubmit, onChange }) => (
  <div className="container post-container">
    <div className="form-group">
      <form className="form-horizontal" onSubmit={onSubmit}>
        <h5>Title: </h5>
        <input
          type="text"
          className="form-control"
          placeholder="What's your programming question. Be specific"
          id="title"
          name="title"
          onChange={onChange}
        />
        <br />
        <h5>Description: </h5>
        <textarea
          rows="7"
          cols="14"
          id="description"
          name="description"
          className="form-control"
          onChange={onChange}
        />
        <br />
        <h5>Tags: </h5>
        <input
          type="text"
          className="form-control"
          placeholder="Atleast one tag"
          name="tags"
          id="tags"
          onChange={onChange}
        />
        <br />
        <button type="submit" className="btn btn-primary" onClick={onSubmit}>Post Your Question </button>
      </form>
    </div>
  </div>
);
Question.propTypes = {
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
};

Question.defaultProps = {
  onChange: () => { },
  onSubmit: () => { },
};
export default Question;
