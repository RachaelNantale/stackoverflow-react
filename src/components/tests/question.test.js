import React from 'react';
import { shallow } from 'enzyme';
import Question from '../Question/question';

it('renders the signup component', () => {
  const QuestionComponent = shallow(<Question />);
  expect(QuestionComponent).toMatchSnapshot();
});
