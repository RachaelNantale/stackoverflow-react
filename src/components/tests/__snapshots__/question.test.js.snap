// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the signup component 1`] = `
<div
  className="container"
>
  <div
    className="form-group"
  >
    <form
      className="form-horizontal"
      onSubmit={[Function]}
    >
      <h5>
        Title: 
      </h5>
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="What's your programming question. Be specific"
        type="text"
      />
      <br />
      <h5>
        Description: 
      </h5>
      <textarea
        className="form-control"
        cols="14"
        id="description"
        name="description"
        onChange={[Function]}
        rows="7"
      />
      <br />
      <h5>
        Tags: 
      </h5>
      <input
        className="form-control"
        id="tags"
        name="tags"
        onChange={[Function]}
        placeholder="Atleast one tag"
        type="text"
      />
      <br />
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
      >
        Post Your Question 
      </button>
    </form>
  </div>
</div>
`;
